### PR TITLE
Email subject saving fix

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@squallstar @tonytlwu 

**Fixed behavior:**
When trying to save email subject and reopen it after the email subject stays the same.

ref Fliplet/fliplet-studio#3590

Change-email-subject-fix

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -181,11 +179,17 @@ $('.show-email-provider').on('click', function() {
...
     data: emailProviderData
   });
+  Fliplet.Studio.emit('widget-save-label-update', {
+    text: 'Save'
+  });
 
   emailProvider.then(function onForwardEmailProvider(result) {
     emailProvider = null;
-    emailProviderResult = result.data;
+    dataSource.definition.validation.email.template = result.data;
```
Comment (squallstar) [inline js/interface.js:188]: Can you elaborate on why these changes are required? I'm confused at why this is necessary. Furthermore, it might fail if one of the properties is missing? e.g. https://github.com/Fliplet/fliplet-widget-validation-manager/blob/0590bda8f1142bb1f28f863954916011823348bd/js/interface.js#L167

Comment (YaroslavOvdii) [reply js/interface.js:188]: @squallstar 
At line 167 there is a check that looks into dataSource and decides to use data form dataSource or default template. If there is no template in dataSource we use the default template and on the contrary when dataSource has the template. I have discovered that dataSource is updated only when you press save in the main frame and that's why we can see only template email subject when we try to go back before saving the main frame. Therefore I decided to update dataSource after closing this provider.
I can see now that not every dataSource have properties which we need. Maybe if I add a check that looks when there are all needed properties in the dataSource object and will add them when they are not, will it solve the issue?

Comment (squallstar) [reply js/interface.js:188]: @YaroslavOvdii looks like you haven't take any action on this yet?
